legacy-common/src/main/java/eu/arrowhead/legacy/common/model/LegacyModelConverter.java

```diff
@@ -112,12 +112,14 @@ public static ServiceRegistryRequestDTO convertLegacyServiceRegistryEntryToServi
 		final ServiceRegistryRequestDTO dto = new ServiceRegistryRequestDTO();
 		dto.setServiceDefinition(entry.getProvidedService().getServiceDefinition());
 		dto.setProviderSystem(provider);
-		dto.setServiceUri(entry.getServiceUri());
+		dto.setServiceUri(entry.getServiceURI());
 		dto.setVersion(entry.getVersion());
 		dto.setSecure(calculateSecurityType(entry.getProvidedService().getServiceMetadata()));
 		entry.getProvidedService().getServiceMetadata().remove(LegacyCommonConstants.KEY_SECURITY);
 		entry.getProvidedService().getServiceMetadata().put(LegacyCommonConstants.KEY_LEGACY_INTERFACE, entry.getProvidedService().getInterfaces().iterator().next());
-		dto.setInterfaces(List.of(LegacyCommonConstants.DEFAULT_INTERFACE));
+		List<String> interfaceList = new ArrayList<String>();
+		interfaceList.addAll(entry.getProvidedService().getInterfaces());
+		dto.setInterfaces(interfaceList);
```

Comment:
As I described previously the reason, the translator won't work properly with this modification and service registry request will be rejected when the interface is described like "XML" instead of "HTTPS-SECURE-XML".